base/channels.jl

```diff
@@ -493,14 +493,18 @@ function show(io::IO, ::MIME"text/plain", c::Channel)
 end
 
 function iterate(c::Channel, state=nothing)
-    try
-        return (take!(c), nothing)
-    catch e
-        if isa(e, InvalidStateException) && e.state === :closed
-            return nothing
-        else
-            rethrow()
+    if isopen(c) && isready(c)
```

Comment:
This will not drain a closed channel right? Should this be `isopen(c) || i ready(c)`